src/repository/UserRepository.java

```diff
@@ -1,28 +1,29 @@
 package repository;
 
 import model.User;
+import modelLayer.Role;
 import util.MyList;
 
 import java.util.concurrent.atomic.AtomicInteger;
 
 public class UserRepository {
 
     private final MyList<User> users;
-
     private final AtomicInteger currentId = new AtomicInteger(1);
 
 
     public UserRepository(MyList<User> users) {
         this.users = users;
     }
 
-    public User addUser(String name, String email,String password){
-        User user = new User(currentId.getAndIncrement(), name, email, password);
+    public User addUser(String name, String email, String password, Role role){
```

Comment:
The role should not be passed to this function. The role will be defined in the user model constructor.